dm/relay/relay_test.go

```diff
@@ -453,14 +455,44 @@ func (t *testRelaySuite) TestHandleEvent(c *C) {
 		c.Assert(errors.Cause(handleErr), Equals, reader2.err)
 	}
 
+	// reader return fake rotate event
+	reader2.err = nil
+	reader2.result.Event = fakeRotateEv
+	// writer return error to force handleEvents return
+	writer2.err = errors.New("writer error for testing")
+	// return with the annotated writer error
+	_, err = r.handleEvents(ctx, reader2, transformer2, writer2)
+	c.Assert(errors.Cause(err), Equals, writer2.err)
+	// after handle rotate event, we save and flush the meta immediately
+	c.Assert(r.meta.Dirty(), Equals, false)
+	{
```

Comment:
just a question, why ues new blocks? seems this is not a common usage, cau you tell me under what circumstances should I use `local code block`😬

Reply:
isolate cases to make it easier to read as we put many cases in a single test method

can also be used to isolate local variables, although not used here